### PR TITLE
Shortcodes to Doctrine [MAILPOET-2993]

### PR DESCRIPTION
**Pull request by pavel-mailpoet**
[MAILPOET-2993]

[MAILPOET-2993]: https://mailpoet.atlassian.net/browse/MAILPOET-2993

### CODE REVIEW

File: tests/integration/Newsletter/ShortcodesTest.php [hunk 1]
```diff
@@ -136,76 +136,53 @@ public function testItCanProcessNewsletterShortcodes() {
   public function itCanProcessPostNotificationNewsletterNumberShortcode() {
     // create first post notification
     $postNotificationHistory = $this->_createNewsletter(
-      $parentId = $this->newsletterId,
+      $parentId = $this->newsletter,
```
Comment (lysyjan) [inline tests/integration/Newsletter/ShortcodesTest.php:145]: This assignment is from old code probably, but it's unnecessary. You used refactor, maybe, and it's the reason why you didn't notice. Similar problem ins on lines 148-149.

**Review by lysyjan — COMMENTED** [review]
Good job! I have a few notes but nothing critical. I hope that I haven't overlooked something important.

File: lib/Newsletter/Shortcodes/Shortcodes.php [hunk 2]
```diff
@@ -5,11 +5,11 @@
 use MailPoet\WP\Functions as WPFunctions;
 
 class Shortcodes {
+  const SHORTCODE_CATEGORY_NAMESPACE = 'MailPoet\Newsletter\Shortcodes\Categories\\';
```
[the remark this answers is not in the document]
Comment (pavel-mailpoet) [reply lib/Newsletter/Shortcodes/Shortcodes.php:16]: I forgot we can start using it for constants. But after I added the visibility I realised this constant is not used any more.

File: tests/integration/Newsletter/ShortcodesHelperTest.php [hunk 3]
```diff
@@ -21,21 +28,18 @@ public function testGetsShortcodes() {
...
+    $shortcodes = $this->shortcodesHelper->getShortcodes();
     expect(count($shortcodes['Subscriber']))->equals(6);
     $customSubscriberShortcode = end($shortcodes['Subscriber']);
-    expect($customSubscriberShortcode['text'])->equals($customField->name);
+    expect($customSubscriberShortcode['text'])->equals($customField->getName());
     expect($customSubscriberShortcode['shortcode'])
-      ->equals('[subscriber:cf_' . $customField->id . ']');
-  }
-
-  public function _after() {
```
[the remark this answers is not in the document]
Comment (pavel-mailpoet) [reply tests/integration/Newsletter/ShortcodesHelperTest.php:38]: I think that if we don't need that we shouldn't keep it, it only slows down the test run.

**Review by lysyjan — APPROVED** [review]

**Review by veljkho — APPROVED** [review]

**Pull request merged**